src/rdb.c

```diff
@@ -1930,6 +1949,13 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid) {
                 }
                 o->type = OBJ_HASH;
                 o->encoding = OBJ_ENCODING_ZIPLIST;
+                if (hashTypeLength(o) == 0) {
+                    zfree(encoded);
+                    o->ptr = NULL;
+                    decrRefCount(o);
+                    goto emptykey;
```

Comment:
Did you consider moving the freeing stuff to `emptykey` (conditionally, if o is not `NULL`)?

Reply:
it's not that simple, each of these `if`s (in all the surrounding code, and especially in streams need to free different things depending on the stage of failure.
in this specific case we at least have to free `encoded` manually.
but if you'll look at other places with failure, ones that `return`, you'll see that each of them frees different things.

we can declare all the different buffers and things to be freed at the top and set them to null, but then we need to nullify them as soon as we use them.
for instance, in stream, we read a group name (string), then we create a group, release the string, then we attach the group to the rax, if we fail before attaching, we need to release the group, and if we fail after attaching, then it's already part of `o`.

point is, that at at this point in time i think we need to stick to what we already have. unless we some day change that entire function and how it handles errors.

Reply:
@yossigo I thought about this at first, and my thoughts were the same as @oranagra.
Moving the freeing stuff to `emptykey` made the code more confusing for me.